runner/src/runner.rs

```diff
@@ -346,17 +352,24 @@ async fn read_chain_storage<T: 'static + Decode + Debug, V: RunnerExt + StorageR
         .map_err(Into::into)
 }
 
-async fn run_binary(runner: &mut impl RunnerExt, stdout_mode: impl Into<Stdio>) -> Result<Child, Error> {
+fn run_binary_with_retry(runner: &mut impl RunnerExt, stdout_mode: impl Into<Stdio>) -> Result<Child, Error> {
     // Ensure there is no other child running
     if runner.child_proc().is_some() {
         return Err(Error::ChildProcessExists);
     }
     let downloaded_release = runner.downloaded_release().as_ref().ok_or(Error::NoDownloadedRelease)?;
-    let child = Command::new(downloaded_release.path.as_os_str())
-        .args(runner.vault_args().clone())
-        .stdout(stdout_mode)
-        .spawn()?;
-    Ok(child)
+    let mut command = Command::new(downloaded_release.path.as_os_str());
+    command.args(runner.vault_args().clone()).stdout(stdout_mode);
+    match retry::<_, _, Child, _>(ExponentialBackoff::default(), || {
```

Comment:
What are the default settings? How long will it retry? I'm thinking it might be appropriate in this case to do a simple loop with a 1 second sleep, and to timeout after e.g. 1 minute. The action that we're doing here is cheap so there's no harm in trying every second.

Reply:
0.5 sec initial sleep, with a timeout after 15 mins (https://github.com/ihrwein/backoff/blob/587e2da8fb2dcfc65ca544cb9249022c51f1406e/src/default.rs#L3).

A loop can also work, or I can just set the `multiplier` to `1` and `initial_interval` to 1 sec. I plan on adding an async retry on the `reqwest::get` to download the vault release, so this crate will still be used.

Reply:
Both approaches work, if we keep `backoff` I would suggest adding a comment for the default configuration. In any case we should eventually timeout if startup is not possible.

Reply:
I'm not against using this crate as such. Using multiplier of 1 with 1 sec would work for me. But what do you guys think of the total timeout period? I feel like for the initial startup, 15 minutes is way too long - I would prefer to give faster feedback to the user that something is not working. When the version changes while running, 15 minutes would be fine

Reply:
@sander2 defined a custom configuration with a one minute timeout